notebooks/tutorials/intro_for_model_developers.ipynb

```diff
@@ -0,0 +1,1688 @@
+    "\n",
+    "Let's train a simple logistic regression model on the dataset and evaluate its performance. You will use the `LogisticRegression` class from the `sklearn.linear_model` and use ValidMind tests to evaluate the model's performance.\n",
+    "\n",
+    "Before training the model, we need to encode the categorical features in the dataset. You will use the `OneHotEncoder` class from the `sklearn.preprocessing` module to encode the categorical features. The categorical features in the dataset are `Geography` and `Gender`.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "metadata": {}
+   },
+   "outputs": [],
+   "source": [
+    "new_df_raw.head()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "metadata": {}
+   },
+   "outputs": [],
+   "source": [
+    "new_df_raw = pd.get_dummies(\n",
+    "    new_df_raw, columns=[\"Geography\", \"Gender\"], drop_first=True\n",
+    ")\n",
+    "new_df_raw.head()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "metadata": {}
+   },
+   "outputs": [],
+   "source": [
+    "from sklearn.linear_model import LogisticRegression\n",
+    "from sklearn.model_selection import train_test_split\n",
+    "\n",
+    "# Split the input and target variables\n",
+    "X = new_df_raw.drop(\"Exited\", axis=1)\n",
+    "y = new_df_raw[\"Exited\"]\n",
+    "X_train, X_test, y_train, y_test = train_test_split(\n",
+    "    X,\n",
+    "    y,\n",
+    "    test_size=0.2,\n",
+    "    random_state=42,\n",
+    ")\n",
+    "\n",
+    "# Logistic Regression grid params\n",
+    "log_reg_params = {\n",
+    "    \"penalty\": [\"l1\", \"l2\"],\n",
+    "    \"C\": [0.001, 0.01, 0.1, 1, 10, 100, 1000],\n",
+    "    \"solver\": [\"liblinear\"],\n",
+    "}\n",
+    "\n",
+    "# Grid search for Logistic Regression\n",
+    "from sklearn.model_selection import GridSearchCV\n",
+    "\n",
+    "grid_log_reg = GridSearchCV(LogisticRegression(), log_reg_params)\n",
+    "grid_log_reg.fit(X_train, y_train)\n",
+    "\n",
+    "# Logistic Regression best estimator\n",
+    "log_reg = grid_log_reg.best_estimator_"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Initialize model evaluation objects and assigning predictions\n",
+    "\n",
+    "The last step for evaluating the model's performance is to initialize the ValidMind `Dataset` and `Model` objects and assign model predictions to each dataset. You will use the `init_dataset`, `init_model` and `assign_predictions` functions to initialize these objects.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "metadata": {}
+   },
+   "outputs": [],
+   "source": [
+    "TRAIN = X_train\n",
```

Comment:
Pedantic comment: Not very important, but just flagging for consistency shake. Should we adhere to a specific naming convention for dataframes, such as using lowercase and appending `_df` to clarify that these are dataframes?

Reply:
Pedantic comment accepted.